### PR TITLE
Progressbar

### PR DESCRIPTION
**Pull request by willroyMyles**

### CODE REVIEW

**Comment by willroyMyles** [comment]
`setTitle(QString string)` - sets the title of the progress bar.
This ProgressBar inherits from QProgressBar so all functions are accessible.

File: src/subclass/progressbar.cpp [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+#include <QDebug>
+#include <QWindow>
+
+ProgressBar::ProgressBar(QWidget *parent)
+	: QProgressBar(parent)
+{
+	ConfigureUI();
+	configureConnection();
+	//show();
+	QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: This isn't needed per class, it's set once in main

File: src/subclass/progressbar.cpp [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+#include <QTime>
+#include <QScreen>
+#include <QDesktopWidget>
+#include <QDebug>
+#include <QWindow>
+
+ProgressBar::ProgressBar(QWidget *parent)
+	: QProgressBar(parent)
+{
+	ConfigureUI();
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: this isn't Java, be consistent with function names

File: src/subclass/progressbar.cpp [hunk 3]
```diff
@@ -0,0 +1,382 @@
+#include "progressbar.h"
+#include <QApplication>
+#include <QGraphicsDropShadowEffect>
+#include <QPainter>
+#include <QPropertyAnimation>
+#include <QShortcut>
+#include <QTime>
+#include <QScreen>
+#include <QDesktopWidget>
+#include <QDebug>
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: Try to remove these from code when you're done

File: src/subclass/progressbar.cpp [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+	proPainter->deleteLater();
+}
+
+QPushButton * ProgressBar::confirmButton()
+{
+	if (confirm) return confirm;
+	return nullptr;
+}
+
+QPushButton * ProgressBar::cancleButton()
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: spelling

File: src/subclass/progressbar.cpp [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+	title = new QLabel(this);
+	content = new QWidget;
+	proPainter = new ProgressPainter;
+	contentLayout = new QVBoxLayout;
+	layout = new QVBoxLayout;
+	closeBtn = new QPushButton(this);
+	confirm = new QPushButton;
+	cancel = new QPushButton;
+	showingCancelDialog = false;
+	confirmationText = "would you like to close the dialog?";
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: wrap strings with `tr()` for future

File: src/subclass/progressbar.cpp [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+	setLayout(layout);
+	layout->setContentsMargins(10, 10, 10, 10);
+	layout->setSpacing(0);
+	layout->addWidget(content);
+
+	content->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+	content->setLayout(contentLayout);
+	content->setStyleSheet("background:rgba(30,30,30,1);");
+
+	title->setText("hi");
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: ??

File: src/subclass/progressbar.cpp [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+}
+
+void ProgressBar::setTitle(QString string)
+{
+	title->setText(string);
+}
+
+void ProgressBar::screenChanged()
+{
+	/*qDebug() << devicePixelRatio();
```
Comment (iKlsR) [inline src/subclass/progressbar.cpp]: remove all comments

**Comment by willroyMyles** [comment]
Do not merge

**Pull request closed without merging**